### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by hueniverse** [TheWidlarzGroup/react-native-video#4944] [Ideas]
### What problem are you trying to solve?

There are currently no tests for this React Native module. The author isn't sure how to write tests for a module like this but assumes it's possible, and would like test coverage to exist.

### Proposed solution

Set up some tests for the module. Anything at all to get test coverage started would be a great first step. A `test` label is ready to be applied.

### Additional context

Original title: "Testing"

Quote from the author: "I have no clue how to write tests for a React Native module like this but I would assume it's possible... Anyone wants to take a first stab at setting up some tests? Anything at all would be fantastic! I am all ready with a `test` label!"

**Comment by freeboub** [comment]
Good question ... I don't think jest is very useful ...
It is possble to add a string describing player state in JS which describe player state, and then test with appium

**Comment by freeboub** [comment]
Good question ... I don't think jest is very useful ...
It is possble to add a string describing player state in JS which describe player state, and then test with appium

**Comment by foloinfo** [comment]
I don't think it's possible to test a native module with jest.

However, you can test if the props are passed / changed correctly after mocking the library.
`jest.mock('react-native-video', ()=> 'ReactNativeVideo')`.

Only problem is I cannot test the behaviour of `ref` s like `ref.seek`.
If anyone has a good solution, I would like to know.

**Comment by talal-tilted** [comment]
Hello did someone find a solution to test it ?

**Comment by karthik-narayan-v** [comment]
By using this below mock we can test react-native-video:

```
jest.mock('react-native-video',()=>{
    const React = require('react');
    class Video extends React.Component {
        constructor(props:any){
            super(props);
            this.seek = this.seek.bind(this);
        }
        seek(){}

        render(){
            return React.createElement('Video',this.props,null);
        }
    }
    return Video;
})
```

**Comment by freeboub** [comment]
@laughingdevil thank you for this peace of code !
In fact there are 2 topics.
- Testing app with jest: your code corresponds to this.
- Testing the package itself with sample data (Todo)

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
